#문제 이해

1로 이루어진 n의 배수를 구하라

예를 들면 111 ---> 3을 나누면 37로 나누어 떨어진다.

#1트 - while 무한루프를 사용해서 해결


In [ ]:
n = int(input())

i = 1
while True:
  if i < n:
    pass
  elif i % n != 0:
    pass
  elif i % n == 0:
    break
  i = str(i) + str(1)
  i = int(i)
a = str(i)
print(a)
a = list(a)
print(type(a))


3
111


TypeError: ignored

In [ ]:
n = int(input())

9901


In [ ]:

i = 1
while True:
  if i < n:
    pass
  elif i % n != 0:
    pass
  elif i % n == 0:
    a = i
    break
  i = str(i) + str(1)
  i = int(i)
result = list(str(a))
print(len(result))
  

TypeError: ignored

틀림---- 아마 이걸 3 7 9901을 한번에 입력할때 답이 나오는 문제를 말하는거 같은데.

#2트 input을 while문 안에 집어 넣어라.

In [ ]:
list = []
i = 1
while 1:
  try:
    n = input()
    if i < n:
      pass
    elif i % n != 0:
      pass
    elif i % n == 0:
      a = i
      break
    i = str(i) + str(1)
    i = int(i)
    result = list(str(a))
    print(len(result))
  except:
    break
print(a)

7
111111111111


In [ ]:
B = []
while 1:
    try:
      A = input()
      B.append(A)
    except:
      break
B = list(map(int,B))
print(B)

3
7
9901
[3, 7, 9901]


In [ ]:
C = []
for i in B:
  j = 1
  while True:
    if j < i:
      pass
    elif j % i !=0:
      pass
    elif j % i == 0:
      C.append(j) 
      break
    j = str(j) + str(j)
    j = int(j)
print(C)

KeyboardInterrupt: ignored

In [ ]:

n = int(input())

a = 0
i = 1
while True:
  if i < n:
    pass
  elif i % n != 0:
    pass
  elif i % n == 0:
    a = i
    break
  i = str(i) + str(1)
  i = int(i)
result = list(str(a))
print(len(result))

7
6


In [ ]:
B = []

while True:
  try:
    A = int(input())
  except:
    break
    if i < A:
      pass
    elif i % A != 0:
      pass
    elif i % n == 0:
      a = i
      break
    i = str(i) + str(1)
    i = int(i)
    B.append(i)
print(B)


3
7
9901

[]


In [1]:
print("1" * 3)

111


#3트 구글링
 

##1번 방법

자문 자답 합니다 sys.stdin.readline은 EOF에러일 떄 공백을 반환한다고 합니다.

따라서 try, except문을 쓸때 except에 eof에러에 대해 넣어 주어야 한다.

In [ ]:
import sys
input = sys.stdin.readline
while(1):
    try:
        a = int(input().rstrip())    
    except KeyboardInterrupt:
        break
    num = 1
    while(1):
        if(num % a == 0):
            print(len(str(num)))
            break
        num = num*10 + 1

내가 못풀었던 이유:

1. i를 더해주는 걸 너무 복잡하게 코드를 짬. 즉 이걸로 시간 복잡도가 올라버림  - 진짜 그냥 "i" * 3만 해도 iii나오는데

2. 너무 많은 if문을 난발함.if문 하나만 써도 적당한데 너무 남발함